pkg/utils/version/version.go

```diff
@@ -104,6 +106,8 @@ func init() {
 	utilruntime.Must(err)
 	ConstraintK8sGreaterEqual125, err = semver.NewConstraint(">= 1.25")
 	utilruntime.Must(err)
+	ConstraintK8sLess125, err = semver.NewConstraint("< 1.25")
```

Comment:
IIRC, such expressions can be problematic, cc @ialidzhikov (same applies for `ConstraintK8sLess124`). Probably you can just use `!ConstraintK8sGreaterEqual125`?

Reply:
I guess you mean the problem and discussion behind https://github.com/gardener/gardener/pull/6209. A problematic constraint is for example `<= 1.25` - does not consider the patch versions of 1.25. `< 1.25` should work.